src/Microsoft.ComponentDetection.Detectors/pub/PubComponentDetector.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.IO;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.ComponentDetection.Contracts;
+using Microsoft.ComponentDetection.Contracts.Internal;
+using Microsoft.ComponentDetection.Contracts.TypedComponent;
+using YamlDotNet.Serialization;
+using YamlDotNet.Serialization.NamingConventions;
+
+namespace Microsoft.ComponentDetection.Detectors.Pub
+{
+    [Export(typeof(IComponentDetector))]
+    public class PubComponentDetector : FileComponentDetector
```

Comment:
New detectors should initially inherit the `IDefaultOffComponentDetector` class until validation can be performed.
```suggestion
    public class PubComponentDetector : FileComponentDetector, IDefaultOffComponentDetector 
```